### PR TITLE
Remove explicit `any` types from the codebase

### LINKED ISSUE #201: remove explicit `any` type annotations in pages code
There are a number of places in `packages/wrangler/pages` that are annotated with `any` (11 as of filing this issue. That makes that code brittle. It's also making our PRs a bit noisy. It should be replaced with better annotations.

### PR DESCRIPTION
**Pull request by petebacondarwin**
This change removes all use of `any` from the code and updates the `no-explicit-any` eslint rule to be an error.

Resolves #201

### CODE REVIEW

File: packages/wrangler/pages/functions/template-worker.ts [hunk 1]
```diff
@@ -1,14 +1,12 @@
...
 import type { HTTPMethod } from "./routes";
 
 /* TODO: Grab these from @cloudflare/workers-types instead */
-type Params<P extends string = any> = Record<P, string | string[]>;
+type Params<P extends string = string> = Record<P, string | string[]>;
 
 type EventContext<Env, P extends string, Data> = {
   request: Request;
-  waitUntil: (promise: Promise<any>) => void;
+  waitUntil: (promise: Promise<unknown>) => void;
```
Comment (threepointone) [inline packages/wrangler/pages/functions/template-worker.ts:9]: I don't think `waitUntil` can return anything, so this should probably be `void`
```suggestion
  waitUntil: (promise: Promise<void>) => void;
```

**Review by threepointone — APPROVED** [review]

File: packages/wrangler/pages/functions/template-worker.ts (hunk 1 above)
Comment (GregBrimble) [reply packages/wrangler/pages/functions/template-worker.ts:9]: It doesn't _do_ anything with the return, but I can still return something if I choose.

Comment (GregBrimble) [reply packages/wrangler/pages/functions/template-worker.ts:9]: `waitUntil(Promise.resolve(5))` is totally valid.

File: packages/wrangler/pages/functions/template-worker.ts [hunk 2]
```diff
@@ -1,14 +1,12 @@
-/* eslint-disable @typescript-eslint/no-explicit-any */
-
 import { match } from "path-to-regexp";
 import type { HTTPMethod } from "./routes";
 
 /* TODO: Grab these from @cloudflare/workers-types instead */
```
Comment (GregBrimble) [inline packages/wrangler/pages/functions/template-worker.ts:4]: As I've said, I do hope to get rid of these entirely soon.

**Review by GregBrimble — APPROVED** [review]

**Review by threepointone — APPROVED** [review]

**Pull request merged**